src/devrules/core/git_service.py

```
@@ -448,3 +417,85 @@ def get_current_repo_name() -> str:
+    """Checkout a branch"""
+    try:
+        subprocess.run(["git", "checkout", selected_branch], check=True)
+        return True, f"Switched to branch '{selected_branch}'"
+    except subprocess.CalledProcessError as e:
+        return False, str(e)
+
+
+def push_branch(branch: str):
+    """Push a branch to remote."""
+    try:
+        subprocess.run(
```

Ruff (0.14.13)

`subprocess` call: check for execution of untrusted input

(S603)